src/main/scala/calc/Compiler.scala

```diff
@@ -74,6 +78,23 @@ object Compiler {
       case Literal(value) =>
         irt.DoubleLiteral(value)
 
+      case BinaryOp(op, rhs, lhs) =>
+        irt.BinaryOp(binaryOpMap(op), compileExpr(rhs), compileExpr(lhs))
+
+      case Let(name, value, body) =>
+        val irtId = irt.Ident(name.name)
+        irt.Block(List(
+          irt.VarDef(irtId, irtpe.DoubleType, false, compileExpr(value)),
+          compileExpr(body))
+        )
+
+      case Ident(name) =>
+        irt.VarRef(irt.Ident(name))(irtpe.DoubleType)
```

Comment:
You probably know, but I'm pointing out that you're assuming, here, that there indeed *exists* a declaration for `name` in the lexical scope. Ideally you should verify that, and report a type error if it's not the case.

Right now, your compiler will happily compile the invalid program, and produce invalid IR for it.